src/backend/distributed/utils/ruleutils_95.c

```diff
@@ -3624,7 +3624,7 @@ get_variable(Var *var, int levelsup, bool istoplevel, deparse_context *context)
 	else
 	{
 		/* System column - name is fixed, get it from the catalog */
-		attname = get_rte_attribute_name(rte, attnum);
+		attname = get_relid_attribute_name(rte->relid, attnum);
```

Comment:
Yeah, I figured out why this change is needed, but I don't really like it being called blindly for all `rte`s here.

The reason why it fails is because Citus uses a function call node to stash some extra information about the entry (shard identifier, for one), so that we can re-use part of the plan.

Rather than changing the `else` block, it would probably be safer to add another `else if` block that has this code and is only called for Citus plans. Lemme look up how to detect that.